opencv.cpp

```diff
@@ -207,3 +209,55 @@ void* opencv_mat_get_data(const opencv_mat mat)
     auto cvMat = static_cast<const cv::Mat*>(mat);
     return cvMat->data;
 }
+
+struct opencv_jpeg_error_mgr {
+    struct jpeg_error_mgr pub;
+    jmp_buf setjmp_buffer;
+};
+
+void opencv_jpeg_error_exit(j_common_ptr cinfo) {
+    opencv_jpeg_error_mgr* myerr = (opencv_jpeg_error_mgr*) cinfo->err;
+    (*cinfo->err->output_message)(cinfo);
+    longjmp(myerr->setjmp_buffer, 1);
```

Comment:
can you explain why a `longjmp` approach is required?

Reply:
It's different, for sure. The `longjmp` function is used along with `setjmp` in libjpeg to handle errors with an immediate jump out of the nested library code back to a safe point (`setjmp`), bypassing standard error flows.
This method is necessary because C lacks built-in exception handling, enabling this code to recover gracefully from errors in JPEG processing.

Reply:
interesting, I see, a quirk of the libjpeg api